deploy/base/gradle-v0.1.yaml

```diff
@@ -0,0 +1,352 @@
+      image: $(params.IMAGE)
+      securityContext:
+        runAsUser: 0
+      env:
+        - name: GRADLE_USER_HOME
+          value: $(workspaces.maven-settings.path)/.gradle
+      workingDir: $(workspaces.source.path)/$(params.CONTEXT_DIR)
+      args: [ "$(params.GOALS[*])" ]
+      script: |
+        #!/usr/bin/env bash
+
+        TOOL_VERSION="$(params.TOOL_VERSION)"
+        [ -n "${TOOL_VERSION}" ] || TOOL_VERSION=$(sed -n "s/distributionUrl=.*\/gradle-\(.*\)-.*\.zip$/\1/p" gradle/wrapper/gradle-wrapper.properties 2>/dev/null)
+
+        if [ -n "${TOOL_VERSION}" ]; then
+            export GRADLE_HOME="/opt/gradle-${TOOL_VERSION}"
+            echo "GRADLE_HOME=${GRADLE_HOME}"
+            case "${TOOL_VERSION}" in
+                7.[3456789]*)
+                    SUPPORTED_JAVA_VERSION="17"
+                    sed -i -e 's|//allowInsecureProtocol|allowInsecureProtocol|g' ${GRADLE_USER_HOME}/init.gradle
+                    ;;
+                7.*)
+                    SUPPORTED_JAVA_VERSION="11"
+                    sed -i -e 's|//allowInsecureProtocol|allowInsecureProtocol|g' ${GRADLE_USER_HOME}/init.gradle
+                    ;;
+                6.*)
+                    SUPPORTED_JAVA_VERSION="11"
+                    ;;
+                5.*)
+                    SUPPORTED_JAVA_VERSION="11"
+                    ;;
+                4.*)
+                    SUPPORTED_JAVA_VERSION="1.8.0"
+                    ;;
+                *)
+                    echo "Unsupported Gradle version: ${TOOL_VERSION}" >&2
+                    exit 1
+                    ;;
+            esac
+        else
+            echo "Could not set Gradle version from gradle/wrapper/gradle-wrapper.properties" >&2
+            exit 1
+        fi
+
+        # XXX: Workaround for com.github.sherter.google-java-format not working on Java 17
+        if find -type f -name 'build.gradle*' -print0 | xargs -0 grep -q com.github.sherter.google-java-format; then
+            echo "Found com.github.sherter.google-java-format plugin. Forcing Java version to 11" >&2
+            SUPPORTED_JAVA_VERSION="11"
+        fi
+
+        JAVA_VERSION="$(params.JAVA_VERSION)"
+
+        [ -n "${JAVA_VERSION}" ] || JAVA_VERSION="${SUPPORTED_JAVA_VERSION}"
+
+        export JAVA_HOME="/usr/lib/jvm/java-${JAVA_VERSION}-openjdk"
+        export PATH="${JAVA_HOME}/bin:${PATH}"
+        export PATH="${GRADLE_HOME}/bin:${PATH}"
+        export LANG=en_US.UTF-8
+        export LC_ALL=en_US.UTF-8
+
+        if [ -n "$(params.ENFORCE_VERSION)" ]; then
+            gradle-manipulator -l "${GRADLE_HOME}" $(params.GRADLE_MANIPULATOR_ARGS) -DversionOverride=$(params.ENFORCE_VERSION) generateAlignmentMetadata || exit 1
+        else
+            gradle-manipulator -l "${GRADLE_HOME}" $(params.GRADLE_MANIPULATOR_ARGS) generateAlignmentMetadata || exit 1
+        fi
+
+        gradle -DAProxDeployUrl=file:$(workspaces.source.path)/hacbs-jvm-deployment-repo $@ || exit 1
```

Comment:
So this is supposed to be the 'user activated' task, it is basically an alternative to the tasks in https://github.com/redhat-appstudio/build-definitions. It should not be deploying the result to the sidecar, rather the real tasks will create a runnable application. I don't want to hold the PR up over this, as it is not really on the critical path at all, but if we actually want to support user level gradle builds then this will need some work and a corresponding build in the build-definitions repo.

Reply:
You can view `gradle -DAProxDeployUrl=file:$(workspaces.source.path)/hacbs-jvm-deployment-repo publish` as equivalent to `mvn -DaltDeploymentRepository=local::file:$(workspaces.source.path)/hacbs-jvm-deployment-repo deploy`.

The equivalent line is still in the Maven version:

```
/usr/bin/mvn -s "$(workspaces.maven-settings.path)/settings.xml" $@ "-DaltDeploymentRepository=local::file:$(workspaces.source.path)/hacbs-jvm-deployment-repo" "org.apache.maven.plugins:maven-deploy-plugin:3.0.0-M2:deploy" 
```